consensus/ibft/src/main/java/org/hyperledger/besu/consensus/ibft/blockcreation/IbftMiningCoordinator.java

```diff
@@ -57,11 +70,36 @@ public IbftMiningCoordinator(
   }
 
   @Override
-  public void enable() {}
+  public void start() {
+    if (started.compareAndSet(false, true)) {
+      ibftExecutors.start();
+      controller.start();
+      ibftExecutors.executeOnSingleThread(ibftProcessor);
+    }
+  }
+
+  @Override
+  public void stop() {
+    if (started.get() && stopped.compareAndSet(false, true)) {
+      ibftProcessor.stop();
+      controller.stop();
+      ibftExecutors.stop();
+    }
+  }
+
+  @Override
+  public void awaitStop() {
+    ibftExecutors.awaitStop();
+  }
+
+  @Override
+  public boolean enable() {
+    return false;
```

Comment:
For the `minerStart` this will now always return false, whereas for the other APIs it returns something sensible.  Could this instead return `started.get() && !stopped.get()` for enable and `!started.get() || stopped.get()` for disable?

Reply:
Well, it appear that the `IbftMiningCoordinator` isn't designed to be enabled / disabled.  It's assumed to always be on.  So any efforts to enable / disable will always fail.  The return value of `false` is meant to indicate that no change was made, which could help a user realize that their request was ineffective (these methods are hooked up to "miner_start" and "miner_stop" JSON-RPC methods). 

This is a modification of the previous behavior where we would shutdown the `IbftMiningCoordinator` when `disable()` was called.  But I'm pretty sure this was a bug, because there was no way to subsequently re-enable the miner.

Reply:
The RPC api that uses this has very poorly documented return code.  Geth's variant returns nothing or an error for enable and disable (https://github.com/ethereum/go-ethereum/wiki/Management-APIs#miner_start).  Our docs say if the request was received successfully, not if it was possible or if a change was made (https://besu.hyperledger.org/en/latest/Reference/API-Methods/#miner_start).  Personally I would expect `setEnabled(true)` to be idempotent and if there is a return to always return the same value on multiple calls.  The biggest problem is we don't have a source of truth to measure what the API should be doing.

Reply:
It's not clear to me which approach is better, since we do have an "eth_mining" command to check the actual mining status.  I reworked as you suggested @shemnon since I think the behavior you're suggesting is closer to the current behavior.  I'll open some tickets to rework the docs and confirm the behavior we want.